### PR TITLE
[#158] Added code for "Create Test" enable/ disable functionality according to test types.

### PR DESCRIPTION
**Pull request by saksham1919**
Create Test button enables/  disables according to test types.

## Description
 The target field is made not required for "Compile", "Memory-Errors" and "Script" test types.

## How has this been tested?
Existing and New RSpec and Cucumber passing.

## Screenshots (if appropriate):
image
image
image

## Checklist:

- [*] Zero `rubocop` violations
- [*] My code receives an "A" when `rubycritic` is run
- [*] >90% test coverage of my changes
- [*] All new and existing tests passed

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -33,6 +33,66 @@
       </div>
     </div>
     <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-YvpcrYf0tY3lHB60NNkmXc5s9fDVZLESaAA55NDzOxhy9GkcIdslK1eN7N6jIeHz" crossorigin="anonymous"></script>
+
```
Comment (masarkar-tamu) [inline app/views/layouts/application.html.erb:36]: Could the following be put in a separate javascript file under `app/javascript`?

**Comment by masarkar-tamu** [comment]
image

Apologies beforehand for being nitpicky, but do you think "create test" button should literally be "grayed" out until all the requisite information is filled in?

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (saksham1919) [reply app/views/layouts/application.html.erb:36]: Sure, done.

**Comment by saksham1919** [comment]
> image
> 
> Apologies beforehand for being nitpicky, but do you think "create test" button should literally be "grayed" out until all the requisite information is filled in?

I think, yes, that's expected. Based on our previous discussions, the "create test" button should indeed remain grayed out until all the required information is completed.

**Review by masarkar-tamu — APPROVED** [review]

**Pull request merged**